ci/gpu/build.sh

```diff
@@ -108,6 +110,11 @@ python setup.py install
 
 cd $WORKSPACE
 
+################################################################################
+# CHECK - Run clang-tidy
+################################################################################
+
+python ./cpp/scripts/run-clang-tidy.py -cdb ./cpp/build/compile_commands.json
```

Comment:
I believe this should be in `ci/checks/style.sh` instead. Can we test running `cmake` in the style job to do the style checks there? I don't think it will add a lot of time to the style check (getting a run will help us know the impact), plus it'll land it with the other style checks to give the user better feedback.

We don't want to fail a GPU test when it's due to style

Reply:
@mike-wendt the issue here is that `clang-tidy` requires a compilation database (list of compilation units in the project and their respective commandlines). This is generated only after running `cmake`. This was the reason I was told by @raydouglass in PR #1931 to move such a check to `ci/gpu/build.sh`.

Reply:
Ray's concern was that you would also be running `make` and building code. If you're doing setup to do a style check it should be in `style.sh`

Reply:
@mike-wendt but the issue is running `cmake ..` inside `style.sh` complains that it cannot find nvcc. So, should I file a PR to our gpuCI container to add cuda toolkit?

Reply:
Can you advice the right way here, @mike-wendt ?

Reply:
@mike-wendt or @raydouglass  - just wanted to reopen this discussion. It seems like we have two possible patterns:

(1) Move this to checks/style.sh: that will require running cmake within style.sh. Unfortunately, that fails on cpu-only instances as nvcc is not installed on them. That will slow down style.sh a bit too.

(2) Keep this here in build.sh: slows down build a little (though not too much: ~ 2min) but can take advantage of existing cmake database that already exists.

It seems like (2) is easiest to integrate with existing CI/build infrastructure, which is why @teju85 went that way, but we can revisit (1) if cmake can be made to work within a check-style job. Unfortunately we're not totally sure how to do that... maybe Ray would know better?

Reply:
You should be able to find `nvcc` in the style job. It runs the same container as the CPU builds. Can you move this to `ci/check/style.sh` and we can see how it goes with CI?

```
$ docker run -it --rm gpuci/rapidsai-base-driver:cuda10.0-ubuntu16.04-gcc5-py3.6 nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Sat_Aug_25_21:08:01_CDT_2018
Cuda compilation tools, release 10.0, V10.0.130
```

Reply:
Ok. I have just pushed a commit with this change. Let's see how it goes.

Reply:
@raydouglass please refer to the failure in the `check-style` CI. Running `cmake` from inside `checks/style.sh` always fails for me with the following error:
```
CMake Error at CMakeLists.txt:21 (project):
  No CMAKE_CUDA_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CUDACXX" or the CMake cache entry CMAKE_CUDA_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.

-- Configuring incomplete, errors occurred!
```

Reply:
I can't reproduce locally, but this line from the CPU build might help: https://github.com/rapidsai/cuml/blob/branch-0.14/ci/cpu/build.sh#L14

Reply:
ok... the env-var certainly helped get past this error. Thanks @raydouglass . However, now, since clang-tidy is almost compiling the code, I'll need all the dependencies of cuML, the ones listed in `cpp/cmake/Dependencies.cmake` to be available!

Reply:
I'm able to make all necessary changes to get clang-tidy running successfully from `style.sh` itself. Unfortunately, since we need to have make all the dependencies available before launching `clang-tidy`, the runtime of `check-style` will now be about 8-9mins (compared to about 2 mins before).

@raydouglass can you review if this change is OK from ops side?